src/core/config.go

```diff
@@ -506,35 +505,70 @@ func (config *Configuration) ContainerisationHash() []byte {
 // GetBuildEnv returns the build environment configured for this config object.
 func (config *Configuration) GetBuildEnv() []string {
 	config.buildEnvStored.Once.Do(func() {
-		env := []string{
-			// Need to know these for certain rules.
-			"ARCH=" + config.Build.Arch.Arch,
-			"OS=" + config.Build.Arch.OS,
-			// These are slightly modified forms that are more convenient for some things.
-			"XARCH=" + config.Build.Arch.XArch(),
-			"XOS=" + config.Build.Arch.XOS(),
-			// It's easier to just make these available for Go-based rules.
-			"GOARCH=" + config.Build.Arch.GoArch(),
-			"GOOS=" + config.Build.Arch.OS,
+		config.buildEnvStored.Env = config.getBuildEnv(true)
+		for _, e := range config.buildEnvStored.Env {
+			if strings.HasPrefix(e, "PATH=") {
+				config.buildEnvStored.Path = strings.Split(strings.TrimPrefix(e, "PATH="), ":")
+			}
 		}
+	})
+	return config.buildEnvStored.Env
+}
+
+// Path returns the slice of strings corresponding to the PATH env var.
+func (config *Configuration) Path() []string {
+	config.GetBuildEnv() // ensure it is initialised
+	return config.buildEnvStored.Path
+}
 
-		// from the BuildEnv config keyword
-		for k, v := range config.BuildEnv {
-			pair := strings.Replace(strings.ToUpper(k), "-", "_", -1) + "=" + v
-			env = append(env, pair)
+func (config *Configuration) getBuildEnv(expanded bool) []string {
+	maybeExpandHomePath := func(s string) string {
+		if !expanded {
+			return s
 		}
+		return ExpandHomePath(s)
+	}
 
-		// from the user's environment based on the PassEnv config keyword
-		for _, k := range config.Build.PassEnv {
-			if v, isSet := os.LookupEnv(k); isSet {
-				env = append(env, k+"="+v)
+	env := []string{
+		// Need to know these for certain rules.
+		"ARCH=" + config.Build.Arch.Arch,
+		"OS=" + config.Build.Arch.OS,
+		// These are slightly modified forms that are more convenient for some things.
+		"XARCH=" + config.Build.Arch.XArch(),
+		"XOS=" + config.Build.Arch.XOS(),
+		// It's easier to just make these available for Go-based rules.
+		"GOARCH=" + config.Build.Arch.GoArch(),
+		"GOOS=" + config.Build.Arch.OS,
+	}
+
+	// from the BuildEnv config keyword
+	for k, v := range config.BuildEnv {
+		pair := strings.Replace(strings.ToUpper(k), "-", "_", -1) + "=" + v
+		env = append(env, pair)
+	}
+
+	// from the user's environment based on the PassEnv config keyword
+	path := false
+	for _, k := range config.Build.PassEnv {
+		if v, isSet := os.LookupEnv(k); isSet {
+			if k == "PATH" {
+				// plz's install location always needs to be on the path.
+				v += ":" + maybeExpandHomePath(config.Please.Location)
```

Comment:
Do we want to ensure plz's path goes first here, to prevent randomly installed binaries with the same name clashing?

Reply:
Good idea!